### PR TITLE
Support  the API of CopyObjectPart

### LINKED ISSUE #817: Support the API of CopyPart
**Is this a BUG REPORT or FEATURE REQUEST?**:

> /kind feature

**Anything else we need to know?**:
CopyPart API: Uploads a part by copying data from an existing object as data source. the requirement is from https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPartCopy.html. Precondition feature for CopyPart is supported in the PR

**Environment**:
- Gelato(release/branch) version:
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by sunfch**
**What this PR does / why we need it**:

Uploads a part by copying data from an existing object as data source when to put object with multipart.  You specify the data source by adding the request header x-amz-copy-source in your request and a byte range by adding the request header x-amz-copy-source-range in your request.  
You must initiate a multipart upload before you can upload any part. 

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #817 

**Special notes for your reviewer**:

**Release note**:

```release-note
Dont support to set acl in http header when put object with multipart now
```

### CODE REVIEW

File: s3/pkg/service/multipart.go [hunk 1]
```diff
@@ -415,6 +415,7 @@ func (s *s3Service) CompleteMultipartUpload(ctx context.Context, in *pb.Complete
 		Type:             ObjectTypeNormal,
 		Tier:             utils.Tier1,
 		Size:             totalSize,
+		Acl:              &pb.Acl{CannedAcl: "private"},
```
Comment (sfzeng) [inline s3/pkg/service/multipart.go]: Hi @sunfch Do we support other ACL settings when CompleteMltipartUpload?

File: s3/pkg/service/multipart.go [hunk 2]
```diff
@@ -567,3 +568,140 @@ func (s *s3Service) ListObjectParts(ctx context.Context, in *pb.ListObjectPartsR
...
+			break
+		default:
+			if srcObject.TenantId != tenantId {
+				err = ErrAccessDenied
+				return err
+			}
+		}
+	}
+
+	_, _, _, err = CheckRights(ctx, srcObject.TenantId)
```
Comment (sfzeng) [inline s3/pkg/service/multipart.go]: As acl is already checked, do we need this CheckRights anymore?

File: s3/pkg/service/multipart.go (hunk 1 above)
Comment (sunfch) [reply s3/pkg/service/multipart.go]: @sfzeng  we still dont support to set acl with http header when to put object with multipart now, so      new object uploaded with multipart only has private acl.  thanks

File: s3/pkg/service/multipart.go (hunk 2 above)
Comment (sunfch) [reply s3/pkg/service/multipart.go]: thanks, will delete the code block

**Pull request merged**